c/array/rotate_array.c

```diff
@@ -1,29 +1,119 @@
-#include <array_utils.h>
+
+#include <stdbool.h>
+#include <assert.h>
+#include <time.h>
+#include "array_utils.h"
 
 #define MAX_SIZE 10
 
+/*
+ *rotate array rotaates the given original array for the given
+ *number of times.
+ *for example 1 2 3 4 5 6 7 8 9 10 is the original array
+ *then it roates it to 9 10 1 2 3 4 5 6 7 8 when number of times
+ *rotate is 2.
+*/
+
+void rotate_array(int *nums, int size, int number_of_times_rotate)
+{
+    for (int count = 0; count < number_of_times_rotate; count++)
```

Comment:
Optimization: Here you are rotating the array multiple times (count). can you do this in one loop instead 2?
If yes, how you will do it?